pina/geometry/union.py

```diff
@@ -0,0 +1,98 @@
+import torch
+from .location import Location
+from ..utils import check_consistency
+from ..label_tensor import LabelTensor
+
+
+class Union(Location):
+    """ PINA implementation of Unions of Domains."""
+
+    def __init__(self, geometries):
+        """ PINA implementation of Unions of Domains.
+
+        :param list geometries: A list of geometries from 'pina.geometry' 
+            such as 'EllipsoidDomain' or 'CartesianDomain'
+
+        :Example:
+            # Create two ellipsoid domains
+            >>> ellipsoid1 = EllipsoidDomain({'x': [-1, 1], 'y': [-1, 1]})
+            >>> ellipsoid2 = EllipsoidDomain({'x': [0, 2], 'y': [0, 2]})
+
+            # Create a union of the ellipsoid domains
+            >>> union = GeometryUnion([ellipsoid1, ellipsoid2])
+
+        """
+        for geometry in geometries:
+            check_consistency(geometry, Location, type(geometry))
+
+        super().__init__()
+        self.geometries = geometries
+
+    def variables(self):
```

Comment:
It's okay to leave it property, I just wanted to understand why. I believe that the user should not be able to change the variables. So yes, let's put property decorator.